E0071 needs to be updated to new format

From: src/test/compile-fail/E0071.rs

Error E0071 needs a span_label, updating it from:

```
error[E0071]: `Foo::FirstValue` does not name a struct or a struct variant
  --> src/test/compile-fail/E0071.rs:14:13
   |
14 |     let u = Foo::FirstValue { value: 0 }; //~ ERROR E0071
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0071]: `u32` does not name a struct or a struct variant
  --> src/test/compile-fail/E0071.rs:15:13
   |
15 |     let t = u32 { value: 4 }; //~ ERROR E0071
   |             ^^^^^^^^^^^^^^^^
```

To:

```
error[E0071]: `Foo::FirstValue` does not name a struct or a struct variant
  --> src/test/compile-fail/E0071.rs:14:13
   |
14 |     let u = Foo::FirstValue { value: 0 }; //~ ERROR E0071
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not a struct

error[E0071]: `u32` does not name a struct or a struct variant
  --> src/test/compile-fail/E0071.rs:15:13
   |
15 |     let t = u32 { value: 4 }; //~ ERROR E0071
   |             ^^^^^^^^^^^^^^^^ not a struct
```

Bonus: The span should underline only the name of the thing that's not a struct rather than the whole expression.  Example:

```
error[E0071]: `u32` does not name a struct or a struct variant
  --> src/test/compile-fail/E0071.rs:15:13
   |
15 |     let t = u32 { value: 4 }; //~ ERROR E0071
   |             ^^^ not a struct
```

This may require additional compiler changes.

--- a/src/librustc_typeck/check/mod.rs
+++ b/src/librustc_typeck/check/mod.rs
@@ -3147,9 +3147,12 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
         };
         if variant.is_none() || variant.unwrap().kind == ty::VariantKind::Tuple {
             // Reject tuple structs for now, braced and unit structs are allowed.
-            span_err!(self.tcx.sess, span, E0071,
-                      "`{}` does not name a struct or a struct variant",
-                      pprust::path_to_string(path));
+            struct_span_err!(self.tcx.sess, path.span, E0071,
+                             "`{}` does not name a struct or a struct variant",
+                             pprust::path_to_string(path))
+                .span_label(path.span, &format!("not a struct"))
+                .emit();
+
             return None;
         }
 

--- a/src/test/compile-fail/E0071.rs
+++ b/src/test/compile-fail/E0071.rs
@@ -11,6 +11,11 @@
 enum Foo { FirstValue(i32) }
 
 fn main() {
-    let u = Foo::FirstValue { value: 0 }; //~ ERROR E0071
-    let t = u32 { value: 4 }; //~ ERROR E0071
+    let u = Foo::FirstValue { value: 0 };
+    //~^ ERROR `Foo::FirstValue` does not name a struct or a struct variant [E0071]
+    //~| NOTE not a struct
+
+    let t = u32 { value: 4 };
+    //~^ ERROR `u32` does not name a struct or a struct variant [E0071]
+    //~| NOTE not a struct
 }

--- a/src/test/compile-fail/trait-as-struct-constructor.rs
+++ b/src/test/compile-fail/trait-as-struct-constructor.rs
@@ -13,4 +13,5 @@ trait TraitNotAStruct {}
 fn main() {
     TraitNotAStruct{ value: 0 };
     //~^ ERROR: `TraitNotAStruct` does not name a struct or a struct variant [E0071]
+    //~| NOTE not a struct
 }

